### PR TITLE
Memory Map performance fixes

### PR DESCRIPTION
**Pull request by aqibisbiqa**
# Description

TLDR;
* Map disabled by default
* Can now enable/disable the map in frontend with a simple switch in Settings pane, disabling massively saves rendering costs
* Disabling in frontend sends signal to agent to not send map updates
* Altered options to enable/disable map for backend
* Can specify how many ticks Craftassist agent sends map updates (default 20 ticks = 1 second)

This PR primarily addresses the drop in performance of the Craftassist agent introduced in my [#1207](https://github.com/facebookresearch/fairo/pull/1207). Essentially, all the socket-io emits from the agent to update the dashboard clog the connection between the front and back end. 

BACKEND: Disabling the map sends a signal to the agent to not send (or calculate) any map updates at all. For craftassist, if the map is enabled, the agent will now only send updates every N ticks (that you can specify). The default is every 20 ticks (1 second). Some of the argument parsing was also modified:
* `python craftassist_agent.py` will run the agent, but the agent won't send any map updates
* `python craftassist_agent.py --draw_map --map_update_ticks 60` will run the agent and send map updates every 60 ticks
* `python locobot_agent.py --draw_map --map_data "memory"` will run the agent and draw the map from memory

FRONTEND: one can disable the memory map entirely with a switch in the settings pane. By default the map is disabled (for both Robot and Craftassist), so you need to click the switch to see anything on the map. If the map doesn't load immediately after clicking the switch, it is possible that the agent hasn't entirely properly loaded. If the map still doesn't load, try clicking the map pane and it should appear. The performance savings here come from the new `enable_map` state variable, which, if false, does an early return of the memory map render and skips all the regular map operations.

Note that to enable the map, you need to do TWO things: use the switch introduced in this PR, and specify the `--draw_map` option. Also, the switch won't even appear if you haven't specified the `--draw_map` option.

Fixes # (issue)

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] Proposes a change (non-breaking change that isn't necessarily a bug)
- [x] Refactor
- [x] New feature (non-breaking change that adds a new functionality)
- [ ] Breaking change (fix or feature that would break some existing functionality downstream)
- [ ] This is a unit test
- [ ] Documentation only change
- [ ] Datasets Release
- [ ] Models Release

## Type of requested review

- [x] I want a thorough review of the implementation.
- [ ] I want a high level review. 
- [ ] I want a deep design review.

## Before and After

You can try running craftassist from `main` right now and see how slow the agent runs when issuing a command. Then, try running from this branch (`aqib/dash_perf_opt`) and you can see the increase in performance when the map is disabled.

**Before:**
Screen Shot 2022-06-13 at 8 47 39 AM

**After:**

Initial View
Screen Shot 2022-06-13 at 8 42 59 AM

After switching on map and clicking on pane
Screen Shot 2022-06-13 at 8 43 13 AM

After switching off map and clicking on pane
Screen Shot 2022-06-13 at 8 43 21 AM

# Testing

See above

# Checklist:

- [x] I have performed manual end-to-end testing of the feature in my environment.
- [x] I have added Docstrings and comments to the code.
- [ ] I have made changes to existing documentation where needed.
- [ ] I have added tests that show that the PR is functional.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added relevant collaborators to review the PR before merge.
- [ ] [Polymetis only] I ran on hardware (1) all scripts in `tests/scripts`, (2) asv benchmarks.

### CODE REVIEW

File: agents/argument_parser.py [hunk 1]
```diff
@@ -100,8 +100,12 @@ def add_mc_parser(self):
         )
         mc_parser.add_argument(
             "--draw_map", 
-            default="memory",
```
Comment (soumith) [inline agents/argument_parser.py:103]: locobot_agent.py also uses this arg, so that has to be updated as well: https://github.com/facebookresearch/fairo/blob/2e00b7b42519e27cfaebd25cec3c6327b79759df/agents/locobot/locobot_agent.py#L268-L273

Comment (soumith) [reply agents/argument_parser.py:103]: maybe a better way would be to keep this option as-is but rename it `map_data`.
Then, also have a flag `draw_map` that either enables it or not.

**Review by soumith — CHANGES_REQUESTED** [review]

File: agents/argument_parser.py (hunk 1 above)
Comment (aqibisbiqa) [reply agents/argument_parser.py:103]: > maybe a better way would be to keep this option as-is but rename it `map_data`. Then, also have a flag `draw_map` that either enables it or not.

Ah ok, that makes sense to break it down. Thanks Soumith!

**Review by soumith — APPROVED** [review]

**Review by kavyasrinet — APPROVED** [review]
Look good after the fixes.
